### DISCUSSION TITLE
`publisher.imageinfo()` in filters

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#402] [Q&A] [answered]
@pgundlach,

I think we already discussed the issue, but I think there could be another approach.

The issue itself is having access to image info data in filters (number of pages, image height and width).

I have just discovered what I think it might be the code for `sd:number-of-pages()`:

https://github.com/speedata/publisher/blob/29ea54b7578b982e3df74d387ff726a052dacee1/src/lua/publisher/layout_functions.lua#L250-L254

Is there no way to make a function similar to the following one available in filters?

``` lua
local function number_of_pages(filename)
  local figure = publisher.imageinfo(filename)
  return figure.img.pages
end
```

If that where possible, this would help to get figure dimensions in filters.

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
This is not (easily) possible, I am afraid. The `number-of-pages()` function uses LuaTeX, whereas during the preprocessing state I don't have access to LuaTeX.

**Reply by pgundlach** [reply]
It could be solved, but a more or less major undertaking.

**Reply by pr-apes** [reply]
In that case, I will try to solve it in a different way.

Many thanks for your help.
